quantum/quantum.h

```diff
@@ -197,6 +197,8 @@ typedef uint8_t pin_t;
 
 #    define readPin(pin) ((bool)(PINx_ADDRESS(pin) & _BV((pin)&0xF)))
 
+#    define togglePin(pin) writePin(pin, !readPin(pin))
```

Comment:
I think the idea is good. But the implementation is not correct.

writePin() and readPin() access different I/O registers.

```suggestion
#    define togglePin(pin) (writePin(pin, !(PORTx_ADDRESS(pin) & _BV((pin)&0xF))))
```

Reply:
macro expansion takes care of using the the correct registers - among the AVRs registers, PORT are meant for writing, and PIN for reading

Reply:
PORT can be read and written. In your implementation, the result of writing to PORT passes through the output circuit and into the input circuit. And reading the input circuit from the PIN.

It works as expected, but it's strictly not good.

I reread the datasheet and found that the AVR has the ability to toggle the output data. It's probably best to use this.

```suggestion
#    define togglePin(pin)  (PINx_ADDRESS(pin) |= _BV((pin)&0xF))
```

Reply:
ah, now i get where you are going with this!
also took a peek into who chibios does its palToggleLine on the avrs - palTogglePort { palWritePort(port,  palReadLatch(port) ^ (bits)) }  with AVRs "pal_lld_readlatch(port) ((port)->out)" = it also uses the output register; so PORT it is then - thanks! :-)

Reply:
https://hackaday.com/2011/07/09/hardware-xor-for-output-pins-on-avr-microcontrollers/